### PR TITLE
[CHORE] Possibility to create environment with system installed uv

### PR DESCRIPTION
**Pull request by maruschin**
I encountered an inconvenient dependency on a pre-installed version of Python.
It is very convenient to create an environment with the required version via uv.

### CODE REVIEW

**Comment by jaychia** [comment]
IIUC this PR means that instead of:

* `uv`-managed Python with a `uv`-managed Python environment

We now allow for:

* system-managed Python that has `uv` installed, with a `uv`-managed Python environment

I'm not the *most* familiar with `uv`, but would it not be the most reproducible developer experience if `uv` were to manage everything, including the Python installation? I know we've had issues in the past with inconsistent system installs of Python.

**Comment by maruschin** [comment]
@jaychia it's most like:
* system-installed `uv` managed Python and Python environment

`uv` does not depend on Python and can be installed on its own.

**Comment by jaychia** [comment]
Makes sense, thanks for the contribution!

File: Makefile [hunk 1]
```diff
@@ -21,17 +22,21 @@ endif
 
 
 .venv:  ## Set up virtual environment
+ifeq (, $(shell which uv))
 	python3 -m venv $(VENV)
 	$(VENV_BIN)/python -m pip install --upgrade uv
-	## Hacks to deal with grpcio compile errors on m1 macs
+else
+	uv venv $(VENV) -p $(PYTHON_VERSION)
```
Comment (maruschin) [inline Makefile:29]: Here I choose which version of python I would use in my venv. And uv just download it and install.

**Pull request merged**